velox/dwio/common/DirectDecoder.h

```diff
@@ -77,6 +78,10 @@ class DirectDecoder : public IntDecoder<isSigned> {
             this->template skip<false>(toSkip, current, nullptr);
           }
           if (atEnd) {
+            if constexpr (Visitor::kHasHook) {
+              visitor.setNumValues(
+                  Visitor::kHasFilter ? numValues : visitor.numRows());
```

Comment:
`kHasHook` and `kHasFilter` cannot be both true.  Also is this file used in Parquet?  If not we don't need to change this file.

Reply:
@Yuhta 
> `kHasHook` and `kHasFilter` cannot be both true

So this is different from [here](https://github.com/facebookincubator/velox/blame/main/velox/dwio/parquet/reader/StringDecoder.h#L79)? I saw it also check both `kHasHook` and `kHasFilter`.

> Also is this file used in Parquet?

Yes, in the ut, if column `b` is `PLAIN`, it will use this, if it's `RLE_DICTIONARY`, it will use `RleBpDataDecoder`, but I didn't test `DeltaBpDecoder`.